src/webgpu/web_platform/reftests/canvas_size_different_with_back_buffer_size.html.ts

```diff
@@ -0,0 +1,208 @@
+      assert(ctx !== null);
+      const width = cvs_larger_than_back_buffer.width / 2;
+      const height = cvs_larger_than_back_buffer.height / 2;
+      ctx.configure({
+        device: t.device,
+        format: 'bgra8unorm',
+        usage: GPUTextureUsage.COPY_DST,
+        size: [width, height, 1], // half of the canvas width and height.
+      });
+
+      const rows = height;
+      const bytesPerRow = 256;
+      const buffer = t.device.createBuffer({
+        mappedAtCreation: true,
+        size: rows * bytesPerRow,
+        usage: GPUBufferUsage.COPY_SRC,
+      });
+      const mapping = buffer.getMappedRange();
+      const data = new Uint8Array(mapping);
+      // 1st row
+      data.set(red, 0);
+      data.set(red, 4);
+      data.set(green, 8);
+
+      // 2nd row
+      data.set(red, 256 + 0);
+      data.set(red, 256 + 4);
+      data.set(green, 256 + 8);
+
+      // 3rd row
+      data.set(blue, 256 * 2 + 0);
+      data.set(blue, 256 * 2 + 4);
+      data.set(yellow, 256 * 2 + 8);
+
+      // 4th row
+      data.set(blue, 256 * 3 + 0);
+      data.set(blue, 256 * 3 + 4);
+      data.set(yellow, 256 * 3 + 8);
+      buffer.unmap();
+
+      const texture = ctx.getCurrentTexture();
+
+      const encoder = t.device.createCommandEncoder();
+      encoder.copyBufferToTexture({ buffer, bytesPerRow }, { texture }, [width, height, 1]);
+      t.device.queue.submit([encoder.finish()]);
+      await t.device.queue.onSubmittedWorkDone();
+    }
+
+    {
+      const ctx = cvs_same_as_back_buffer.getContext('gpupresent');
+      assert(ctx !== null);
+      ctx.configure({
+        device: t.device,
+        format: 'bgra8unorm',
+        usage: GPUTextureUsage.COPY_DST,
+      });
+
+      const rows = cvs_same_as_back_buffer.height;
+      const bytesPerRow = 256;
+      const buffer = t.device.createBuffer({
+        mappedAtCreation: true,
+        size: rows * bytesPerRow,
+        usage: GPUBufferUsage.COPY_SRC,
+      });
+      const mapping = buffer.getMappedRange();
+      const data = new Uint8Array(mapping);
+      // 1st row
+      data.set(red, 0); // red
+      data.set(red, 4); // red
+      data.set(green, 8); // green
+
+      // 2nd row
+      data.set(red, 256 + 0); // red
+      data.set(red, 256 + 4); // red
+      data.set(green, 256 + 8); // blue
+
+      // 3rd row
+      data.set(blue, 256 * 2 + 0); // blue
+      data.set(blue, 256 * 2 + 4); // blue
+      data.set(yellow, 256 * 2 + 8); // red
+
+      // 4th row
+      data.set(blue, 256 * 3 + 0); // blue
+      data.set(blue, 256 * 3 + 4); // blue
+      data.set(yellow, 256 * 3 + 8); // yellow
+      buffer.unmap();
+
+      const texture = ctx.getCurrentTexture();
+
+      const encoder = t.device.createCommandEncoder();
+      encoder.copyBufferToTexture({ buffer, bytesPerRow }, { texture }, [3, 4, 1]);
+      t.device.queue.submit([encoder.finish()]);
+      await t.device.queue.onSubmittedWorkDone();
```

Comment:
nit: the onSubmittedWorkDone calls shouldn't be needed

Reply:
Adding this previously to ensure when the test achieve to the end, the canvas should present things. But agree to remove it.

Reply:
Any DOM changes in the test should appear on the screen in the reftest capture - and that includes drawing to canvases like we do here. (In practice I think the reftest frameworks wait a few frames before grabbing a capture.)